tests/wpt/tests/quirks/line-height-preserved-segment-break.html

```diff
@@ -0,0 +1,27 @@
+<!-- quirks mode -->
+<link rel="author" title="Martin Robinson" href="mrobinson@igalia.com">
+<link rel="help" href="https://quirks.spec.whatwg.org/#the-line-height-calculation-quirk">
+<link rel="help" href="https://drafts.csswg.org/css-text/#valdef-white-space-pre-line">
+<link rel="match" href="line-height-preserved-segment-break-ref.html">
+<p>Test passes if there is a filled green square and <strong>no red</strong>.</p>
+
+<style>
+    .container {
+        width: 100px;
+        background: green;
+        white-space: pre-line;
+    }
+
+    .inline-box {
+        line-height: 100px;
+    }
+</style>
+
+<!-- This div is just to make sure that red shows up if the test div doesn't
+     display properly. -->
+<div style="position: absolute; z-index: -100; width: 100px; height: 100px; background: red;"></div>
```

Comment:
Fine either way, but you could turn this into a wrapper to avoid `position: absolute; z-index: -100`:

```html
<div style="width: 100px; height: 100px; background: red;">
  <div class="container"><span class="inline-box">&#xA;</span></div>
</div>
```

I also think the HTML looks slightly visually clearer with `&#xA;` rather than starting one line with `<div>` and next line with `</span>`. But again, not a big deal.